spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1270,7 +1270,9 @@ protected void pollAndInvoke() {
 			}
 			debugRecords(records);
 			resumeConsumerIfNeccessary();
-			resumePartitionsIfNecessary();
+			if (this.consumerPaused && !isPaused()) {
+				resumePartitionsIfNecessary();
+			}
```

Comment:
My comment on the issue was wrong, this should be `if (!this.consumerPaused)`.

Reply:
Yes, you're right, I didn't pay enough attention yesterday, it's fixed.

Reply:
Thanks; I see some other issues while writing a test for this; I'll take it from here.

Thanks again for the contribution.